scripts/zones/Bastok_Markets/npcs/Ardea.lua

```diff
@@ -12,21 +12,28 @@ entity.onTrade = function(player, npc, trade)
 end
 
 entity.onTrigger = function(player, npc)
-    local chasingQuotasStatus = player:getCharVar("ChasingQuotas_Progress")
+    local rockRacketeer = player:getQuestStatus(xi.quest.log_id.WINDURST, xi.quest.id.windurst.ROCK_RACKETEER)
 
-    if chasingQuotasStatus == 3 then
-        player:startEvent(264) -- Someone was just asking about that earring.
-    elseif chasingQuotasStatus == 4 then
-        player:startEvent(265) -- They'll be happy if you return it.
+    -- Rock Racketeer
+    if
+        rockRacketeer == QUEST_ACCEPTED and
+        player:hasKeyItem(xi.ki.SHARP_GRAY_STONE)
+    then
+        player:startEvent(261)
     end
 end
 
 entity.onEventUpdate = function(player, csid, option)
 end
 
 entity.onEventFinish = function(player, csid, option)
-    if csid == 264 then
-        player:setCharVar("ChasingQuotas_Progress", 4)
+    -- Rock Racketeer
+    if csid == 261 and option ~= 1 then
+        player:delKeyItem(xi.ki.SHARP_GRAY_STONE)
+        player:addGil(xi.settings.main.GIL_RATE * 10)
+        player:setCharVar("rockracketeer_sold", 1)
+    elseif csid == 261 and option ~= 2 then
+        player:setCharVar("rockracketeer_sold", 2)
```

Comment:
Lol what's going on here?

Reply:
some other quest that i didnt touch that that NPC is a part of - -